contents/code/tilingmanager.js

```diff
@@ -299,16 +299,16 @@ function TilingManager(timer) {
                               "Meta+Shift+M",
                               function() {
                                   try {
-                                      var layout = self.layouts[workspace.currentDesktop - 1][workspace.activeScreen];
-                                      if (layout != null) {
-                                          var client = workspace.activeClient;
-                                          if (client != null) {
-                                              var tile = layout.getTile(client.x, client.y);
+                                      var client = workspace.activeClient;
+                                      if (client != null) {
+                                          var tile = self.tiles.getTile(client);
+                                          if (tile != null) {
+                                              var layout = self.layouts[tile.getDesktop() - 1][tile.getScreen()];
```

Comment:
(I hate these different desktop numbering schemes!)

Reply:
Yes, confusing and error-prone.